src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftPluginDescriptor.java

```diff
@@ -61,13 +61,13 @@ default ListBoxModel doFillWaitUnitItems() {
     default FormValidation doTestConnection(@QueryParameter String apiURL, @QueryParameter String authToken) {
 
 
-        if (StringUtils.isEmpty(apiURL)) {
-            if(!EnvVars.masterEnvVars.containsKey(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY) &&
-                    !StringUtils.isEmpty(EnvVars.masterEnvVars.get(IOpenShiftPlugin.KUBERNETES_SERVICE_HOST_ENV_KEY))) {
+        if (apiURL == null || StringUtils.isEmpty(apiURL)) {
+            if(!EnvVars.masterEnvVars.containsKey("KUBERNETES_MASTER") &&
```

Comment:
So to record my review comment in line, still check for KUBERNETES_SERVICE_HOST first (and add KUBERNETES_SERVICE_PORT ), then if that does not exist, try KUBERNETES_MASTER